zulip_bots/zulip_bots/bots/link_shortener/test_link_shortener.py

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+
+from zulip_bots.test_lib import BotTestCase
+
+class TestMessageInfoBot(BotTestCase):
+    bot_name = "link_shortener"
+
+    def test_bot(self):
+        with self.mock_config_info(
+            { 'key': '<your API key>' }
+        ):
+            self.initialize_bot()
+
+        message = 'Shorten https://www.github.com/zulip/zulip please.'
+        response = 'https://www.github.com/zulip/zulip: https://goo.gl/6uoWKb'
```

Comment:
Is the bot making an http request to the API here? You should stub that request as done in the tests for giphy bot.

Reply:
No request to the API is being made on lines 14 and 15. Since goo.gl always returns the same shortened link for the same url (as long as the REST API is being used), https://www.github.com/zulip/zulip will always be converted to https://goo.gl/6uoWKb. Because of this, `test_bot` just tests if given that `message`, the `response` is `'https://www.github.com/zulip/zulip: https://goo.gl/6uoWKb'`.

I'm not sure what "stubbing" is, can you please explain it? I looked in the giphy test and did a search for "stub" but there were no results. I also looked at some articles online, and they mentioned using "mock configurations," but I'm not entirely sure how to use those either. Thank you very much!

Reply:
Stubbing is replacing an actual function with a fake function so that we can test it in isolation. See the call to `mock_http_conversation` in the giphy tests. Also, look at the `fixtures/*.json` files that show mock http conversations.

So, yes, the line 14-15 are making a request to the API but getting the same result back. This means that we can use this result to create a mock http conversation now